File: Tarumt.CC.Ecommerce.SharedLibrary/Infrastructure/Responses/ProductResponse.cs
namespace Tarumt.CC.Ecommerce.SharedLibrary.Infrastructure.Responses
{
    public class ProductResponse : Response
    {
        public required string Name { get; set; }

        public required string ShortName { get; set; }

        public required int Count { get; set; }

        public required decimal Price { get; set; }

        public required decimal DiscountRate { get; set; }

        public required string Description { get; set; }

        public required string ImageUrl { get; set; }

        public required string[] CategoriesId { get; set; }

        public DateTime? StartAt { get; set; }

        public DateTime? ExpiredAt { get; set; }
    }
}

File: Tarumt.CC.Ecommerce.FE/src/app/types/responses/product-response.ts
import { ApiResponseBase } from "./api-response.response";

export interface ProductResponse extends ApiResponseBase {
  name: string;
  short_name: string;
  count: number;
  price: number;
  discount_rate: number;
  description: string;
  image_url: string;
  categories_id: string[];
  start_at: string;
  expired_at: string;
}
